libmcount/mcount.c

```diff
@@ -1745,8 +1751,10 @@ static void mcount_cleanup(void)
 	mcount_finish();
 	destroy_dynsym_indexes();
 
+#ifndef SINGLE_THREAD
 	pthread_key_delete(mtd_key);
 	mtd_key = -1;
+#endif
```

Comment:
Yes these changes are correct but it hurts readability of the code IMO.  It has no runtime cost in the fast path and only consumes a tiny bit of memory so why not leave it as is?

Reply:
Okay. I will remove it.